libtransact/src/scheduler/serial/core.rs

```diff
@@ -202,46 +328,47 @@ impl SchedulerCore {
                     self.try_schedule_next()?;
                 }
                 Ok(CoreMessage::ExecutionResult(task_notification)) => {
-                    let batch = match self.current_batch.take() {
-                        Some(batch) => batch,
-                        None => {
-                            panic!("received execution result but no current batch is executing")
-                        }
-                    };
-
-                    let results = match task_notification {
-                        ExecutionTaskCompletionNotification::Valid(context_id) => {
+                    let current_txn_id = self.current_txn.as_ref().ok_or_else(|| {
+                        CoreError::InternalError(
+                            "received execution result but no current transaction is executing"
+                                .into(),
+                        )
+                    })?;
+                    match task_notification {
+                        ExecutionTaskCompletionNotification::Valid(context_id, transaction_id) => {
+                            if &transaction_id != current_txn_id {
```

Comment:
We should make this an error callback (maybe in a separate PR).

Reply:
This will be addressed in an error callback PR